src/datatypes.ts

```diff
@@ -153,6 +156,110 @@ export function deserializeDatetime(literalString: string): Date | null {
   return date;
 }
 
+/**
+ * Time type for time data type attributes
+ */
+export type Time = {
+  hour: number;
+  minute: number;
+  second: number;
+  millisecond?: number;
+  timezoneHourOffset?: number;
+  timezoneMinuteOffset?: number;
+};
+
+/**
+ * @internal
+ * @param value Value to serialise.
+ * @returns String representation of `value` in UTC.
+ * @see https://www.w3.org/TR/xmlschema-2/#time-lexical-representation
+ */
+export function serializeTime(value: Time): string {
+  const utcHours =
+    value.hour + (value.timezoneHourOffset ? value.timezoneHourOffset : 0);
+  const utcMinutes =
+    value.minute +
+    (value.timezoneMinuteOffset ? value.timezoneMinuteOffset : 0);
```

Comment:
Hmm, by normalising everything to UTC, we're throwing away information. For example, if I set my alarm to 8:00, I don't want that to suddenly become 7:00 when Daylight Saving Time starts (or ends, whichever is appropriate :P ). So I think rather than adding the offsets to the hour and minute values, it's probably best to explicitly set the offset to the given values, or `Z` if none are passed.

Reply:
Yep, fair point. I was trying to match the standards we have for Date and DateTime.

Reply:
Yeah that makes sense, unfortunately in the case of `Date` we're forced to throw away that info because Date doesn't allow us to represent timezones, but since we have our own data structure here we can avoid that :)